app/src/ui/preferences/appearance.tsx

```diff
@@ -112,15 +81,52 @@ export class Appearance extends React.Component<
       )
     }
 
+    const darkThemeImage = encodePathAsUrl(__dirname, 'static/ghd_dark.svg')
+    const lightThemeImage = encodePathAsUrl(__dirname, 'static/ghd_light.svg')
+
     return (
       <DialogContent>
-        <Row>
-          <VerticalSegmentedControl
-            items={themes}
-            selectedKey={selectedTheme}
-            onSelectionChanged={this.onSelectedThemeChanged}
-          />
-        </Row>
+        <h2 id="theme-heading">Theme</h2>
+        <RadioGroup<ApplicationTheme>
+          ariaLabelledBy="theme-heading"
+          className="theme-selector"
+          selectedKey={selectedTheme}
+          radioButtonKeys={[
+            ApplicationTheme.Light,
+            ApplicationTheme.Dark,
+            ...(supportsSystemThemeChanges() ? [ApplicationTheme.System] : []),
+            ...(enableHighContrastTheme()
+              ? [ApplicationTheme.HighContrast]
+              : []),
+          ]}
+          onSelectionChanged={this.onSelectedThemeChanged}
+        >
+          <span>
+            <img src={lightThemeImage} alt="" />
+            <span className="theme-value-label">Light</span>
+          </span>
+          <span>
+            <img src={darkThemeImage} alt="" />
+            <span className="theme-value-label">Dark</span>
+          </span>
+          {supportsSystemThemeChanges() && (
+            <span>
+              <span className="system-theme-swatch">
+                <img src={lightThemeImage} alt="" />
+                <img src={lightThemeImage} alt="" />
```

Comment:
Do we need this image twice here? 🤔

Reply:
Yes. It is another trick, the first image is position relatively so that the label grows to the size of the image and the others are absolute so they can be clipped (and overlay the relatively positioned one).

Reply:
Just added some documentation.